channeldb/migration30/migration.go

```diff
@@ -23,12 +23,26 @@ import (
 // indexes.
 const recordsPerTx = 20_000
 
+// MigrateRevLogConfig can be used to configure the MigrateRevocationLog
+// migration function.
+type MigrateRevLogConfig struct {
```

Comment:
This seems pretty rigid, given that afaict we'd pass the _same_ config in for all the migration instances? What about this declaring a specific config interface: 
```
type MigrateRevLogConfig interface {
    NoAmountData(mgrNumber int32) bool 
}
```

Then the "top level" migration config can embed/implement a super set of the config needed by all nodes. The top level type can also stay as `any`, or we could get more involved and try to bind the type as a compile time type parameter.

Reply:
Not entiiiiirely sure what you mean here. Especially re the `mgrNumber` parameter. 

I have, however, now updated the PR to have a new superset `MigrationConfig` struct which houses all the various migration configs (only one for now). I think this makes things slightly nicer. But dont think it is what you meant ..?

Reply:
might also be worth noting that this is currently only ever passed to 1 migration (since it is only passed to optional migrations)

Reply:
just pushed an update that makes the config way more generic (kudos to @guggero  for the idea!)